Crash when using client certifiactes

Using the May 17th release. Tried with key in pk8 and regular pkcs12

./pulsarctl -s https://pulsar-admin-proxy:8443/ --auth-params "tlsCertFile:/pulsar/pulsar-admin.pem,tlsKeyFile:/pulsar/pulsar-admin.key.pem" --auth-plugin "org.apache.pulsar.client.impl.auth.AuthenticationTls" topics list tenant/ns

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x6c2e66]

goroutine 1 [running]:
github.com/streamnative/pulsarctl/pkg/auth.(*TLSAuthProvider).configTLS(0xc00032de00, 0xc00032de00, 0x2)
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/pkg/auth/tls.go:86 +0x126
github.com/streamnative/pulsarctl/pkg/auth.NewAuthenticationTLS(0x7ffeea46cd36, 0x18, 0x7ffeea46cd5a, 0x1c, 0xa92ec0, 0xe08c00, 0xc0003431a0, 0x2, 0x2)
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/pkg/auth/tls.go:44 +0x9c
github.com/streamnative/pulsarctl/pkg/auth.NewAuthenticationTLSFromAuthParams(0x7ffeea46cd2a, 0x4c, 0xa92ec0, 0xe08c00, 0x2, 0xc0003599f0, 0x81b06a)
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/pkg/auth/tls.go:64 +0x1b0
github.com/streamnative/pulsarctl/pkg/auth.GetAuthProvider(0xc000359ae0, 0xc00032ddd0, 0xc000359a00, 0x1)
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/pkg/auth/auth_provider.go:47 +0x348
github.com/streamnative/pulsarctl/pkg/pulsar.New(0xc000359ae0, 0x0, 0x7ffeea46cdcf, 0x4, 0xc000343001)
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/pkg/pulsar/admin.go:77 +0x11e
github.com/streamnative/pulsarctl/pkg/cmdutils.(*ClusterConfig).Client(0xc00001ee60, 0x1, 0xc000343080, 0x0)
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/pkg/cmdutils/config.go:194 +0xfa
github.com/streamnative/pulsarctl/pkg/cmdutils.NewPulsarClient(...)
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/pkg/cmdutils/cmdutils.go:87
github.com/streamnative/pulsarctl/pkg/ctl/topic.doListTopics(0xc00008d720, 0x7ffeea46cdc6, 0xd)
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/pkg/ctl/topic/list.go:82 +0x98
github.com/streamnative/pulsarctl/pkg/ctl/topic.ListTopicsCmd.func1(0xc000210f00, 0xc000359cf8)
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/pkg/ctl/topic/list.go:65 +0x2a
github.com/streamnative/pulsarctl/pkg/cmdutils.run(0xc0001ecd00)
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/pkg/cmdutils/verb.go:100 +0x27
github.com/streamnative/pulsarctl/pkg/cmdutils.(*VerbCmd).SetRunFuncWithNameArg.func1(0xc00021aa00, 0xc00031a310, 0x1, 0x7)
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/pkg/cmdutils/verb.go:75 +0xb3
github.com/spf13/cobra.(*Command).execute(0xc00021aa00, 0xc0001a0c00, 0x7, 0xc, 0xc00021aa00, 0xc0001a0c00)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000ad680, 0xc000342e80, 0xc00001e0b0, 0x9)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
        /home/runner/work/streamnative-ci/streamnative-ci/pulsarctl/main.go:41 +0x1b8

--- a/pkg/auth/auth_provider.go
+++ b/pkg/auth/auth_provider.go
@@ -62,10 +62,6 @@ func GetAuthProvider(config *common.Config) (*Provider, error) {
 
 func getDefaultTransport(config *common.Config) http.RoundTripper {
 	transport := http.DefaultTransport.(*http.Transport)
-	if !config.TLSAllowInsecureConnection && len(config.TLSTrustCertsFilePath) == 0 {
-		return transport
-	}
-
 	tlsConfig := &tls.Config{
 		InsecureSkipVerify: config.TLSAllowInsecureConnection,
 	}

--- a/pkg/ctl/cluster/test.go
+++ b/pkg/ctl/cluster/test.go
@@ -71,7 +71,7 @@ var (
 	basePath string
 )
 
-func TestTLSHelp(newVerb func(cmd *cmdutils.VerbCmd), args []string) (out *bytes.Buffer, err error) {
+func TestTLSHelp(newVerb func(cmd *cmdutils.VerbCmd), args []string) (out *bytes.Buffer, execErr, err error) {
 	var rootCmd = &cobra.Command{
 		Use:   "pulsarctl [command]",
 		Short: "a CLI for Apache Pulsar",
@@ -82,31 +82,31 @@ func TestTLSHelp(newVerb func(cmd *cmdutils.VerbCmd), args []string) (out *bytes
 		},
 	}
 
+	cmdutils.ExecErrorHandler = func(err error) {
+		execErr = err
+	}
+
 	if !flag {
 		basePath, err = os.Getwd()
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
 		err = os.Chdir("../../../")
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
 		basePath, err = os.Getwd()
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		flag = true
 	}
 
 	baseArgs := []string{
-		"--auth-params",
-		"{\"tlsCertFile\":\"" + basePath + "/test/auth/certs/client-cert.pem\"" +
-			",\"tlsKeyFile\":\"" + basePath + "/test/auth/certs/client-key.pem\"}",
-		"--tls-trust-cert-path", basePath + "/test/auth/certs/cacert.pem",
 		"--admin-service-url", "https://localhost:8443",
-		"--tls-allow-insecure"}
+	}
 
 	buf := new(bytes.Buffer)
 	rootCmd.SetOut(buf)
@@ -123,5 +123,5 @@ func TestTLSHelp(newVerb func(cmd *cmdutils.VerbCmd), args []string) (out *bytes
 	cmdutils.AddVerbCmd(flagGrouping, resourceCmd, newVerb)
 	rootCmd.PersistentFlags().AddFlagSet(cmdutils.PulsarCtlConfig.FlagSet())
 	err = rootCmd.Execute()
-	return buf, err
+	return buf, execErr, err
 }

--- a/pkg/ctl/cluster/tls_test.go
+++ b/pkg/ctl/cluster/tls_test.go
@@ -20,6 +20,7 @@
 package cluster
 
 import (
+	"os"
 	"strings"
 	"testing"
 
@@ -27,22 +28,48 @@ import (
 )
 
 func TestTLS(t *testing.T) {
-	args := []string{"clusters", "add", "tls"}
-	_, err := TestTLSHelp(CreateClusterCmd, args)
+	// There is no tls configuration, the execErr should not nil
+	args := []string{"cluster", "add", "tls"}
+	_, execErr, _ := TestTLSHelp(CreateClusterCmd, args)
+	assert.NotNil(t, execErr)
+
+	basePath, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	// The allowInsecureConnection is not specified. So the test should failed with 'doesn't contain any IP SANs'
+	args = []string{
+		"--auth-params",
+		"{\"tlsCertFile\":\"" + basePath + "/test/auth/certs/client-cert.pem\"" +
+			",\"tlsKeyFile\":\"" + basePath + "/test/auth/certs/client-key.pem\"}",
+		"cluster", "add", "tls"}
+	_, execErr, _ = TestTLSHelp(CreateClusterCmd, args)
+	assert.NotNil(t, execErr)
+
+	allArgs := []string{
+		"--auth-params",
+		"{\"tlsCertFile\":\"" + basePath + "/test/auth/certs/client-cert.pem\"" +
+			",\"tlsKeyFile\":\"" + basePath + "/test/auth/certs/client-key.pem\"}",
+		"--tls-trust-cert-path", basePath + "/test/auth/certs/cacert.pem",
+		"--tls-allow-insecure",
+	}
+	args = append(allArgs, []string{"clusters", "add", "tls"}...)
+	_, _, err = TestTLSHelp(CreateClusterCmd, args)
 	assert.Nil(t, err)
 
-	args = []string{"clusters", "list"}
-	out, err := TestTLSHelp(ListClustersCmd, args)
+	args = append(allArgs, []string{"clusters", "list"}...)
+	out, _, err := TestTLSHelp(ListClustersCmd, args)
 	assert.Nil(t, err)
 	clusters := out.String()
 	assert.True(t, strings.Contains(clusters, "tls"))
 
-	args = []string{"clusters", "delete", "tls"}
-	_, err = TestTLSHelp(deleteClusterCmd, args)
+	args = append(allArgs, []string{"clusters", "delete", "tls"}...)
+	_, _, err = TestTLSHelp(deleteClusterCmd, args)
 	assert.Nil(t, err)
 
-	args = []string{"clusters", "list"}
-	out, err = TestTLSHelp(ListClustersCmd, args)
+	args = append(allArgs, []string{"clusters", "list"}...)
+	out, _, err = TestTLSHelp(ListClustersCmd, args)
 	assert.Nil(t, err)
 	clusters = out.String()
 	assert.False(t, strings.Contains(clusters, "tls"))

